src/main/frontend/app/List.js

```diff
@@ -1,39 +1,82 @@
 import React, {Component} from 'react';
+import SideBar from './components/SideBar';
 
 
 class List extends Component {
     constructor(props){
         super(props)
+        let params = (new URL(document.location)).searchParams
+        let budget = params.get("budget")
+        let style = params.get("style")
         this.state = {
-            fashionItems:[]
+            fashionItems:[],
+            budget:budget,
+            style: style
         }
     }
 
     componentDidMount(){
-        fetch("/api/v1/fashion")
-        .then(resp => {
-            return resp.json()
-        })
-        .then(body => {
-            this.setState({fashionItems:body.indexItemsDTOS}) 
-        })
+        if(!this.state.style && !this.state.budget){
+            fetch("/api/v1/fashion")
+            .then(resp => {
+                return resp.json()
+            })
+            .then(body => {
+                this.setState({fashionItems:body.indexItemsDTOS}) 
+            })
+        }
+        else if(this.state.budget && this.state.style){
+            let path = "/api/v1/fashion/?budget=" + this.state.budget + '&style=' + this.state.style 
+            fetch(path)
+            .then(resp => {
+                return resp.json()
+            })
+            .then(body => {
+                this.setState({fashionItems:body.indexItemsDTOS}) 
+            })
+        }
+        else if(this.state.budget){
+            let path = "/api/v1/budget/?budget=" + this.state.budget
+            fetch(path)
+            .then(resp => {
+                return resp.json()
+            })
+            .then(body => {
+                this.setState({fashionItems:body.indexItemsDTOS}) 
+            })
+        }
+        else {
+            let path = "/api/v1/style/?style=" + this.state.style 
+            fetch(path)
+            .then(resp => {
+                return resp.json()
+            })
+            .then(body => {
+                this.setState({fashionItems:body.indexItemsDTOS}) 
+            })
+        }
```

Comment:
Please refactor so the fetch is only written once and the path is a variable set in the conditionals